state_transfer/transfer/transfer.go

```diff
@@ -11,17 +11,21 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+// Transfer knows how to transfer PV data from a source to a destination
 type Transfer interface {
+	// Source returns a source client
 	Source() *rest.Config
+	// Destination returns a destination client
 	Destination() *rest.Config
+	// Endpoint returns the endpoint used by the transfer
 	Endpoint() endpoint.Endpoint
+	// Transport returns the transport used by the transfer
 	Transport() transport.Transport
-	// TODO: define a more generic type for auth
-	// for example some transfers could be authenticated by certificates instead of username/password
-	Username() string
```

Comment:
why remove them without a proper replacement? 

I think some consumers need to access username and password, we should be sure we don't long it otherwise keeping there here doesnt hurt, IMO